spec/index.bs

```diff
@@ -840,6 +881,66 @@ enum GPUBindingType {
+
+<div algorithm="GPUDevice.createBindGroupLayout(descriptor)">
+The <dfn-type="method" dfn-for="GPUDevice">createBindGroupLayout(|descriptor|)</dfn> method is used to create {{GPUBindGroupLayout}}s.
+
+1. Let |layout| be a new valid {{GPUBindGroupLayout}} object.
+1. For each {{GPUBindGroupLayoutBinding}} |bindingDescriptor| in |descriptor|.{{GPUBindGroupLayoutDescriptor/bindings}}:
+    1. Let |layoutBinding| be a new internal [=layout binding=] object.
+    1. If |bindingDescriptor|.{{GPUBindGroupLayoutBinding/type}} is {{GPUBindingType/uniform-buffer}}:
+        1. Set |layoutBinding|.{{layout binding/[[hasDynamicOffsets]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/hasDynamicOffsets}}.
+    1. If |bindingDescriptor|.{{GPUBindGroupLayoutBinding/type}} is {{GPUBindingType/storage-buffer}} or {{GPUBindingType/readonly-storage-buffer}}:
+        1. Set |layoutBinding|.{{layout binding/[[hasDynamicOffsets]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/hasDynamicOffsets}}.
+    1. If |bindingDescriptor|.{{GPUBindGroupLayoutBinding/type}} is {{GPUBindingType/sampled-texture}} or {{GPUBindingType/storage-texture}}:
+        1. Set |layoutBinding|.{{layout binding/[[textureDimension]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/textureDimension}}.
+        1. Set |layoutBinding|.{{layout binding/[[textureComponentType]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/textureComponentType}}.
+        1. Set |layoutBinding|.{{layout binding/[[multisampled]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/multisampled}}.
+    1. Set |layoutBinding|.{{layout binding/[[binding]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/binding}}.
+    1. Set |layoutBinding|.{{layout binding/[[visibility]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/visibility}}.
+    1. Set |layoutBinding|.{{layout binding/[[type]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/type}}.
+    1. Insert |layoutBinding| into |layout|.{{GPUBindGroupLayout/[[bindingsMap]]}} with a key of |layoutBinding|.{{layout binding/[[binding]]}}.
+1. Return |layout|.
+
+<b>Validation</b>
+* The device must not be lost.
+* Each |bindingDescriptor|.{{GPUBindGroupLayoutBinding/binding}} in |descriptor| must be unique.
+* There must be {{GPULimits/maxUniformBuffersPerShaderStage|GPULimits.maxUniformBuffersPerShaderStage}} or 
+    fewer |bindingDescriptor|s of type {{GPUBindingType/uniform-buffer}} visible on each shader stage in |descriptor|.
+    * There must be {{GPULimits/maxDynamicUniformBuffersPerPipelineLayout|GPULimits.maxDynamicUniformBuffersPerPipelineLayout}} or
+        fewer |bindingDescriptor|s of type {{GPUBindingType/uniform-buffer}} with {{GPUBindGroupLayoutBinding/hasDynamicOffsets}} set to `true` 
+        visible across <b>all</b> shader stages in |descriptor|.
```

Comment:
Does one buffer bound to both VERTEX and FRAGMENT count as 1 or 2? (Clarify in text)

here and below